extensions/influxdb/deployment/src/main/java/org/apache/camel/quarkus/component/influxdb/deployment/InfluxdbProcessor.java

```diff
@@ -0,0 +1,76 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.influxdb.deployment;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.IndexDependencyBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageProxyDefinitionBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import org.influxdb.dto.QueryResult;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+
+class InfluxdbProcessor {
+
+    private static DotName INFLUXDB_DTO_PACKAGE = DotName.createSimple("org.influxdb.dto");
+
+    private static final String FEATURE = "camel-influxdb";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    void sslSupport(BuildProducer<ExtensionSslNativeSupportBuildItem> extensionSslNativeSupport) {
+
+        // Indicates that this extension would like the SSL support to be enabled
+        extensionSslNativeSupport.produce(new ExtensionSslNativeSupportBuildItem(FEATURE));
+    }
+
+    @BuildStep
+    void clientProxies(BuildProducer<NativeImageProxyDefinitionBuildItem> proxies) {
+        proxies.produce(new NativeImageProxyDefinitionBuildItem("org.influxdb.impl.InfluxDBService"));
+    }
+
+
+    @BuildStep
+    IndexDependencyBuildItem registerDependencyForIndex() {
+        return new IndexDependencyBuildItem("org.influxdb", "influxdb-java");
+    }
+
+    @BuildStep
+    ReflectiveClassBuildItem registerForReflection(CombinedIndexBuildItem combinedIndex) {
+//        IndexView index = combinedIndex.getIndex();
+//
+//        String[] dtos = index.getKnownClasses().stream()
+//                .filter(ci -> ci.name().prefix().equals(INFLUXDB_DTO_PACKAGE))
+//                .map(ci -> ci.name().toString())
+//                .sorted()
+//                .toArray(String[]::new);
+//
+//        return new ReflectiveClassBuildItem(true, true, dtos);
```

Comment:
@ppalaga , @lburgazzoli 
If I understand your previous comments, this should be the desired solution - do not name classes by one, but use the whole package name.
Unfortunately this is not working in native mode.
In JVM mode I can see that search on the index is correct and returns:

> [org.influxdb.dto.BatchPoints, org.influxdb.dto.BoundParameterQuery, org.influxdb.dto.Point, org.influxdb.dto.Pong, org.influxdb.dto.Query, org.influxdb.dto.QueryResult]

Is it possible that index is not updated by the new build item from previous step (in native), because the desired classes are added in previous step into index?

Reply:
Strange. You may want to add `.peek(System.out::println)` before `.filter()` to see whether the inner classes are in the index. If they appear in the log, that probably means that `ci.name().prefix()` returns something else than you expect.

Reply:
btw the search should not appear neither in jvm nor in native mode as it happens during augmentation

Reply:
I've tried peek, as suggested - but in console there are noin influxdb classes present.
I was able to debug this method in JVM mode -> to be sure, that filter works correctly and it is.
If I disable step with influxdb adition in jvm mode, it behaves in the same way as native (no ivfluxdb classes found) -> IMO index is not updated by added dependency of influxdb. (which could be caused, that I haven't defined any kind of dependency between these 2 steps )

Reply:
you have a dependency as you require CombinedIndexBuildItem which of course need to collect all the IndexDependencyBuildItem.

Reply:
Problem was caused by the fact, that same required classes are definned as inner classes therefore they have different prefix (e.g. org.influxdb.dto.QueryResult$Series). It works with modification of filter to use 'startsWith' instead of 'equals'.